public/app/features/variables/editor/VariableEditorList.tsx

```diff
@@ -1,67 +1,98 @@
-  onDuplicateVariable = (event: MouseEvent, identifier: VariableIdentifier) => {
-    event.preventDefault();
-    this.props.onDuplicateVariable(identifier);
-  };
-
-  onRemoveVariable = (event: MouseEvent, identifier: VariableIdentifier) => {
-    event.preventDefault();
-    this.props.onRemoveVariable(identifier);
+export function VariableEditorList({
+  variables,
+  usages,
+  usagesNetwork,
+  onChangeOrder,
+  onAdd,
+  onEdit,
+  onDelete,
+  onDuplicate,
+}: Props): ReactElement {
+  const onDragEnd = (result: DropResult) => {
+    if (!result.destination || !result.source) {
+      return;
+    }
+    const identifier = JSON.parse(result.draggableId);
+    onChangeOrder(identifier, result.source.index, result.destination.index);
   };
 
-  render() {
-    return (
+  return (
+    <div>
       <div>
-        <div>
-          {this.props.variables.length === 0 && (
-            <div>
-              <EmptyListCTA
-                title="There are no variables yet"
-                buttonIcon="calculator-alt"
-                buttonTitle="Add variable"
-                infoBox={{
-                  __html: ` <p>
+        {variables.length === 0 && <EmptyVariablesList onAdd={onAdd} />}
+
+        {variables.length > 0 && (
+          <div>
+            <table
+              className="filter-table filter-table--hover"
+              aria-label={selectors.pages.Dashboard.Settings.Variables.List.table}
+            >
+              <thead>
+                <tr>
+                  <th>Variable</th>
+                  <th>Definition</th>
+                  <th colSpan={5} />
+                </tr>
+              </thead>
+              <DragDropContext onDragEnd={onDragEnd}>
+                <Droppable droppableId="variables-list" direction="vertical">
+                  {(provided) => (
+                    <tbody ref={provided.innerRef} {...provided.droppableProps}>
+                      {variables.map((variable, index) => (
+                        <VariableEditorListRow
+                          index={index}
+                          key={`${variable.name}-${index}`}
+                          variable={variable}
+                          usageTree={usages}
+                          usagesNetwork={usagesNetwork}
+                          onDelete={onDelete}
+                          onDuplicate={onDuplicate}
+                          onEdit={onEdit}
+                        />
+                      ))}
+                      {provided.placeholder}
+                    </tbody>
+                  )}
+                </Droppable>
+              </DragDropContext>
+            </table>
+          </div>
+        )}
+      </div>
+    </div>
+  );
+}
+
+function EmptyVariablesList({ onAdd }: { onAdd: () => void }): ReactElement {
+  return (
+    <div>
+      <EmptyListCTA
+        title="There are no variables yet"
+        buttonIcon="calculator-alt"
+        buttonTitle="Add variable"
+        infoBox={{
+          __html: ` <p>
```

Comment:
this is a bit iffy, but I think separately we can remove/replace this infoBox prop with one that accepts ReactNode instead.

Thankfully this isn't in grafana-ui :)

Reply:
WOw, I didn't even realize that. I'm not going to change that in this PR.